pennylane/data/utils.py

```diff
@@ -203,122 +121,134 @@ def _check_data_exist(data_type, description, directory_path):
+    if not response.ok:
+        response.raise_for_status()
+
+    global _data_struct
+    _data_struct = response.json()
+
+
+def _s3_download(data_type, folders, attributes, dest_folder):
+    """Download a file for each attribute from each folder to the specified destination."""
+    for folder in folders:
+        s3_folder = os.path.join(S3_URL, data_type, folder)
+        local_folder = os.path.join(dest_folder, data_type, folder)
+        if not os.path.exists(local_folder):
+            os.makedirs(local_folder)
+
+        file_prefix = folder.replace("/", "_")
+        # TODO: if len(attributes) > 1, merge contents to single file like _partial.dat
+        for attr in attributes:
+            fname = f"{file_prefix}_{attr}.dat"
+            response = requests.get(f"{s3_folder}/{fname}", timeout=5.0)
+            if not response.ok:
+                response.raise_for_status()
+            with open(f"{local_folder}/{fname}", "wb") as f:
+                f.write(response.content)
+
+
+def _generate_folders(node, folders):
+    """Recursively generate and return a tree of all folder names below a node.
+
+    Args:
+        node (dict): A sub-dict of the foldermap for which we will generate a list of sub-folders.
+        folders: (list[list[str]]): The ordered list of folder names requested.
+            The value ``["full"]`` will expand to all possible folders at that depth
+
+    Returns:
+        list[str]: The paths of files that should be fetched from S3
+    """
+
+    next_folders = folders[1:]
+    folders = set(node) if folders[0] == ["full"] else set(folders[0]).intersection(set(node))
+    if not next_folders:
+        return folders
+    return [
+        os.path.join(folder, child)
+        for folder in folders
+        for child in _generate_folders(node[folder], next_folders)
+    ]
+
+
+def load(data_type, attributes=None, lazy=False, folder_path="", force=False, **params):
     r"""Downloads the data if it is not already present in the directory and return it to user as a Datset object
 
     Args:
         data_type (str):  A string representing the type of the data required - qchem or qspin
-        description (dict): A dictionary with parameters for the required type of data
         attributes (list): An optional list to specify individual data element that are required
         folder_path (str): Path to the root folder where download takes place. By default dataset folder will be created in the working directory.
         force (Bool): Bool representing whether data has to be downloaded even if it is still present
+        params (kwargs): Keyword arguments exactly matching the parameters required for the data type. Note that these are not optional
 
     Returns:
         list[DatasetFile]
 
     """
 
-    _validate_params(data_type, description, attributes)
-
-    description = {
-        key: (val if isinstance(val, list) else [val]) for (key, val) in description.items()
-    }
+    _ = lazy
```

Comment:
🤔

Reply:
just to avoid linting errors until we actually use `lazy`